src/app-layer-htp-range.c

```diff
@@ -444,12 +491,36 @@ File *ContainerUrlRangeClose(HttpRangeContainerBlock *c, uint16_t flags)
 
     if (f->size >= c->container->totalsize) {
         // we finished the whole file
-        ContainerUrlRangeFileClose(c->container, flags);
+        HttpRangeFileClose(c->container, flags);
     } else {
         // we are expecting more ranges
-        THashDecrUsecnt(c->container->hdata);
         f = NULL;
+        SCLogDebug("expecting more use_cnt %u", SC_ATOMIC_GET(c->container->hdata->use_cnt));
     }
-    SCMutexUnlock(&c->container->mutex);
+    SCLogDebug("returning f %p (c:%p container:%p)", f, c, c->container);
     return f;
 }
+
+static void HttpRangeBlockDerefContainer(HttpRangeContainerBlock *b)
+{
+    if (b && b->container) {
+        DEBUG_VALIDATE_BUG_ON(SC_ATOMIC_GET(b->container->hdata->use_cnt) == 0);
+        THashDecrUsecnt(b->container->hdata);
+        DEBUG_VALIDATE_BUG_ON(SC_ATOMIC_GET(b->container->hdata->use_cnt) > (uint32_t)INT_MAX);
```

Comment:
Is this debug validation useful ? (the one just above looks clearly useful to me)

Reply:
I added it when debugging an issue where I decremented more than incremented. I agree it seems redundant.